application/controllers/admin/themes.php

```diff
@@ -172,6 +172,16 @@ public function tmp($id)
      */
     public function upload()
     {
+        // Code for backward compatiblity with custom themes that are expecting to upload images to upload subaction. 
+        // That happens as their options.twig are outdated.
+        // Now that the upload subaction doesn't handle all uploads, weneed to dispatch to the proper.
+        $action = returnGlobal('action');
+        if ($action == 'templateuploadimagefile') {
```

Comment:
Mersonnaly , i prefer this : 
```` 
        if ($action == 'templateuploadimagefile') {
            return $this->templateuploadimagefile();
        }
        if ($action == 'templateupload') {
            return $this->templateupload();
        }
        // else : continue
````
More clear

Reply:
Done

Reply:
Thanks